plugin/pkg/shoot/validator/admission.go

```diff
@@ -357,6 +368,21 @@ func (c *validationContext) validateScheduling(a admission.Attributes, seedListe
 	return nil
 }
 
+func getNumberOfShootsOnSeed(shootLister corelisters.ShootLister, seedName string) (int64, error) {
+	allShoots, err := shootLister.Shoots(metav1.NamespaceAll).List(labels.Everything())
+	if err != nil {
+		return 0, fmt.Errorf("could not list all shoots: %w", err)
+	}
+
+	count := int64(0)
+	for _, shoot := range allShoots {
+		if seed := shoot.Spec.SeedName; seed != nil && *seed == seedName {
```

Comment:
Do we also need to consider `shoot.Status.SeedName`? cc @plkokanov @kris94

Reply:
Since we already add shoots in deletion to the count we can also check `shoot.Status.SeedName` for shoots that are in migration and add them as well. 
Generally migration should complete faster than deletion, but the shoot could get stuck in an error loop during migration.

Reply:
Ok, but then the scheduler should consider this as well.
I will open a dedicated PR for this :)